### PR TITLE
US9: Admin Group Allocations backend

### PR DESCRIPTION
**Pull request by pharapeti**
## What is this change doing?

- Rearrange Admin groups routes to make more sense
- Provide endpoints to allow Admin to view, create and delete Group Allocations

## Why is it needed?

Admin needs to be able to organise Groups in a Project

## Where should the reviewer start?

server.js
routes/admin/projects/group_allocations/group_allocations.routes.js 
then remaining changes 

## How can these changes be manually tested?

See Postman

## What user story does this PR relate to? (include link)

US9

### CODE REVIEW

File: controllers/admin/projects/group_allocations/group_allocations.controller.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  })
+  .catch(err => {
+    res.status(500).send({message: err.message || `Error occurred retrieving Groups for Project ${req.params.project_id}`});
+  })
+}
+
+// Create a Group Allocation associated to this Project
+exports.createOne = (req, res) => {
+  if (req.body.group_id && req.body.external_id) {
+    model.User.findOne({ where: { external_id: req.body.external_id }}).then(user => {
```
Comment (DaGev) [inline controllers/admin/projects/group_allocations/group_allocations.controller.js:28]: There is no constraint that there can't be 2 users with the same `external_id` so I think this should be changed to just `id` at all the places unless we want to add that as a constraint on the model (which I think would be a better idea given how much `external_id` has been used.

Comment (DaGev) [reply controllers/admin/projects/group_allocations/group_allocations.controller.js:28]: Also, right now, a user can be added to a group multiple times. You can have a group of 5 with all 5 being the same guy. We can either fix this or frankly just ignore it and never demonstrate that in the final presentation, they will have no way of knowing, or we can just leave to resolve this at the end and write it down in the ISSUES section of the repo.

**Review by DaGev — COMMENTED** [review]
Everything works, but I left some comments on the code of potential bugs/improvements. Check it out and say what you think should be done.

File: controllers/admin/projects/group_allocations/group_allocations.controller.js (hunk 1 above)
Comment (pharapeti) [reply controllers/admin/projects/group_allocations/group_allocations.controller.js:28]: >  write it down in the ISSUES section of the repo

Yep please do that, just copy your comments across

**Comment by pharapeti** [comment]
@DaGev Let's just create an issue to resolve the points you mentioned

**Review by DaGev — APPROVED** [review]

**Pull request merged**